crates/fluvio-smartengine/src/transforms/mod.rs

```diff
@@ -18,29 +18,32 @@ mod instance {
     };
 
     use super::{
-        filter::SmartModuleFilter, map::SmartModuleMap, filter_map::SmartModuleFilterMap,
-        array_map::SmartModuleArrayMap, aggregate::SmartModuleAggregate,
+        simple_transform::SimpleTansform, aggregate::SmartModuleAggregate,
+        simple_transform::SimpleTansformKind,
     };
 
     pub(crate) fn create_transform(
         ctx: &SmartModuleInstanceContext,
         initial_data: SmartModuleInitialData,
         store: &mut impl AsContextMut,
     ) -> Result<Box<dyn DowncastableTransform>> {
-        if let Some(tr) = SmartModuleFilter::try_instantiate(ctx, store)?
+        if let Some(tr) = SimpleTansform::try_instantiate(SimpleTansformKind::Filter, ctx, store)?
```

Comment:
Since these are not dynamic, maybe create static instead of enum?

Reply:
Sorry I didn't get it. :eyes: I think enum is also just a set of static values. BTW, since the `kind` is just for getting `name`, I think it doesn't differ a lot for different choices.

Reply:
Enum is probably overkill and not much add value since it is only used once.  In that case it could be just a const:
```

const MAP_FN: &str = "map";
const FILTER_FN: &str = "filter";

...
SimpleTransform: try_instantiate(FILTER_FN,ctx).....

```

Otherwise good simplification

Reply:
Ok. Updated